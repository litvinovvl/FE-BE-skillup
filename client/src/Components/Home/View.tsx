import React from "react";
import Paper from "@material-ui/core/Paper";
import { createStyles, withStyles, WithStyles } from "@material-ui/core/styles";

interface IHomeProps extends WithStyles<typeof styles> {}

const Home: React.SFC<IHomeProps> = ({ classes: { container, heading } }): JSX.Element => {
  const FE = ["Typescript", "React + Redux", "Formik"];
  const BE = ["Node JS", "Typescript", "Express", "Postgresql + Typeorm"];
  const tests = ["Jest + Enzyme", "Mocha", "Cypress"];

  return (
    <Paper className={container}>
      <h1 className={heading}>This is the spring skillup project.</h1>
      <p>This project was written to learn a several front-end and back-end libraries.</p>
      <span>
        <h4 className={heading}>For FE part is used:</h4>
        <ul>
          {FE.map((title, index) => (<li key={index}>{title}</li>))}
        </ul>
        <h4 className={heading}>For BE part is used:</h4>
        <ul>
          {BE.map((title, index) => (<li key={index}>{title}</li>))}
        </ul>
        <h4 className={heading}>For tests is used:</h4>
        <ul>
          {tests.map((title, index) => (<li key={index}>{title}</li>))}
        </ul>
      </span>
      <p>Use top navigation to go to Podcasts page and look how this stack works.</p>
    </Paper>
  )
}

const styles = createStyles({
  container: {
    width: "50%",
    marginTop: "20px",
    margin: "auto",
    padding: "30px"
  },
  heading: {
    color: "#3f51b5"
  }
});

export default withStyles(styles)(Home);
