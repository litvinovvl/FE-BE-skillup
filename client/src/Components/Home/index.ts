import Home from "./View";

export default Home;
