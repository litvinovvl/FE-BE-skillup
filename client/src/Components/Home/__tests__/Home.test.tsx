import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import Home from '../View';

describe('Home component', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<Home />);
  });

  it('should exists', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
})

export {
  // Use an empty export to please Babel's single file emit.
  // https://github.com/Microsoft/TypeScript/issues/15230
}