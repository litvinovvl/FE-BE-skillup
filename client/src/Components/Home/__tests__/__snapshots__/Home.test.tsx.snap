// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
"<WithStyles(Home)>
  <Home classes={{...}}>
    <WithStyles(ForwardRef(Paper)) className=\\"Home-container-1\\">
      <ForwardRef(Paper) classes={{...}} className=\\"Home-container-1\\">
        <div className=\\"MuiPaper-root MuiPaper-elevation1 Home-container-1 MuiPaper-rounded\\">
          <h1 className=\\"Home-heading-2\\">
            This is the spring skillup project.
          </h1>
          <p>
            This project was written to learn a several front-end and back-end libraries.
          </p>
          <span>
            <h4 className=\\"Home-heading-2\\">
              For FE part is used:
            </h4>
            <ul>
              <li>
                Typescript
              </li>
              <li>
                React + Redux
              </li>
              <li>
                Formik
              </li>
            </ul>
            <h4 className=\\"Home-heading-2\\">
              For BE part is used:
            </h4>
            <ul>
              <li>
                Node JS
              </li>
              <li>
                Typescript
              </li>
              <li>
                Express
              </li>
              <li>
                Postgresql + Typeorm
              </li>
            </ul>
            <h4 className=\\"Home-heading-2\\">
              For tests is used:
            </h4>
            <ul>
              <li>
                Jest + Enzyme
              </li>
              <li>
                Mocha
              </li>
              <li>
                Cypress
              </li>
            </ul>
          </span>
          <p>
            Use top navigation to go to Podcasts page and look how this stack works.
          </p>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </Home>
</WithStyles(Home)>"
`;
